packages/lib/src/ContentContainer.wc.svelte

```diff
@@ -0,0 +1,104 @@
+<svelte:options customElement={{ tag: 'sesamy-content-container', shadow: 'open' }} />
+
+<script lang="ts">
+  import type { SesamyAPI, Entitlement } from '@sesamy/sesamy-js';
+  import Base from './Base.svelte';
+  import type { ContentContainerProps } from './types';
+  let {
+    'item-src': itemSrc = '',
+    'publisher-content-id': publisherContentId,
+    'access-url': accessUrl = '',
+    pass = '',
+    'access-level': accessLevel = 'entitlement',
+    'lock-mode': lockMode = 'embed'
+  }: ContentContainerProps = $props();
+
+  async function checkAccess(api: SesamyAPI) {
+    switch (accessLevel) {
+      case 'public':
+        return true;
+      case 'logged-in':
+        return api.auth.isAuthenticated();
+      case 'entitlement':
+      default:
+        return api.entitlements.hasAccess(itemSrc, pass.split(','));
+    }
+  }
+
+  // TODO: This should be moved to sesamy-js
+  async function getRemoteContent(api: SesamyAPI, entitlement: boolean | Entitlement) {
+    if (!accessUrl) {
+      return '';
+    }
+
+    const access =
+      typeof entitlement === 'object' ? await api.entitlements.access(entitlement.id) : undefined;
+
+    const url = new URL(accessUrl);
+    const headers = new Headers({
+      accept: 'text/html'
+    });
+
+    if (access?.token) {
+      headers.set('authorization', `Bearer ${access.token}`);
+    }
+    if (access?.url) {
+      headers.set('x-sesamy-signed-url', access.url);
+    }
+
+    const response = await fetch(url, { headers });
+    const content = await response.text();
+
+    // Parse content and look for the div[slot="content"] on the <sesamy-content-container> tag
+    const parser = new DOMParser();
+    const contentSlot = parser
```

Comment:
This currently fetches the unlocked version of the pages and the corresponding html element. We should support a few other options here, but maybe that should be handled by the content proxy instead?